server/src/modules/mixin/exchange/exchange-client.controller.ts

```diff
@@ -29,9 +30,18 @@ export class ExchangeUserController {
   @ApiOperation({ summary: 'Get order details by id' })
   @ApiResponse({ status: 200, description: 'Get order details by id' })
   @ApiResponse({ status: 400, description: 'Bad Request' })
-  async getDepositAddress(@Param('order_id') orderId: string) {
+  async getDepositAddress(@Param('order_id') orderId: string): Promise<SpotOrderDetails> {
     try {
-      return this.exchagneService.readOrderById(orderId);
+      return {
+        ...(await this.exchagneService.readOrderById(orderId)),
+        amount: `${Math.random()}`, // TODO determine where should it come from
```

Comment:
@zed-wong I'm not quite sure where to get this data from.

Reply:
But front-end part of this task is ready. :) .

Reply:
@mj-blockydevs  I suppose we will need to work this with https://github.com/Hu-Fi/Mr.Market/issues/82 together.

For these values:
```
amount: We will need to add `limitAmount` in `spot-order.entity.ts`

price: We won't need this field, we display price when state is success, and calculate it based on `receive/pay`

avg: We won't need this field

filled: We will need to add `limitFilled` in `spot-order.entity.ts`, display when order is limit, for market it would be same as `receiveAmount`

pay: `amount` in `spot-order.entity.ts`

fee: We won't need this field, calculate it based on `amount * fee (0.2%)`

receive:We will need add `receiveAmount` in `spot-order.entity.ts`, and let state updater(placedOrderUpdater) in `exchange.service.ts` also update this field when success.
```

Reply:
@zed-wong and what is the difference between filled and received for limit order?

Reply:
> @zed-wong and what is the difference between filled and received for the limit order?

Filled is the amount of tokens for limit orders that partially traded on exchange.
Received is the amount of token user finally receive on mixin.

Reply:
Ok, now I get it. But shouldn't I be tracking it in Mixin.listener.ts then?

Reply:
> @zed-wong Ok, now I get it. But shouldn't I be tracking it in Mixin.listener.ts then?

Yes we can do that

Reply:
@zed-wong 
This pull request contains changes for both tasks: 82 and 75. I wasn't sure about the fee calculations.

Is this:
```
 we set the estimated price to subtract 0.2% fees (which in practice is to add 0.2%)
 ```
 
going to be enough? Shouldn't I add these fee somewhere else, for example add it to the payment amount we're sending to Mixin?

Reply:
> @zed-wong This pull request contains changes for both tasks: 82 and 75. I wasn't sure about the fee calculations.
> 
> Is this:
> 
> ```
>  we set the estimated price to subtract 0.2% fees (which in practice is to add 0.2%)
> ```
> 
> going to be enough? Shouldn't I add these fee somewhere else, for example add it to the payment amount we're sending to Mixin?

I missed it, you are right, we should calculate the payment amount based on the estimated price.

Reply:
@zed-wong sure, I'll take care of it. One more thing. For now in "BUY" tab we are typing the amount of USD we are willing to spend to purchase automatically calculated number bitcoins. Shouldn't it be the other way around? Shouldn't we type the number of Bitcoins we want to purchase in this field and then automatically calculate the paymentAmount based on that?

Reply:
> @zed-wong sure, I'll take care of it. One more thing. For now in "BUY" tab we are typing the amount of USD we are willing to spend to purchase automatically calculated number bitcoins. Shouldn't it be the other way around? Shouldn't we type the number of Bitcoins we want to purchase in this field and then automatically calculate the paymentAmount based on that?


Exchanges like Binance and okx they use USDT by default, they also offer switch to type BTC though. I think the current method is alright, for buy we type USDT and for sell we type BTC. Both are what user is going to pay. maybe we can add the switch in the future.